File: utils/password.php
import random

lettres_majuscules = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
lettres_minuscules = "abcdefghijklmnopqrstuvwxyz"
chiffres = "123456789"
caracteres_speciaux = "!@#$%^&*()-=+[]{|;:,.<>?/"

print("Faites votre choix : " )
print("1 - Caractères alphabétiques uniquement")
print("2 - Caractères alphabétiques et numériques")
print("3 - Caractères alphabétiques, numériques et caractères spéciaux")
choix = input("Votre choix (1, 2 ou 3) : ")

longueur_mot_de_passe = 15

groupes = []

if choix == "1":
    groupes = [lettres_majuscules, lettres_minuscules]
elif choix == "2":
    groupes = [lettres_majuscules, lettres_minuscules, chiffres]
elif choix == "3":
    groupes = [lettres_majuscules, lettres_minuscules, chiffres, caracteres_speciaux]
else:
    print("Choix invalide ! Veuillez réessayer.")
    exit()

mot_de_passe = [random.choice(groupe) for groupe in groupes]
characteres_utilises = ''.join(groupes)
mot_de_passe += random.choices(characteres_utilises, k=longueur_mot_de_passe - len(mot_de_passe))

random.shuffle(mot_de_passe)

print("Mot de passe généré :", ''.join(mot_de_passe))

File: utils/password.py
import random

uppercase_letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
lowercase_letters = "abcdefghijklmnopqrstuvwxyz"
digits = "123456789"
special_characters = "!@#$%^&*()-=+[]{|;:,.<>?/"

print("Faites votre choix : " )
print("1 - Caractères alphabétiques uniquement")
print("2 - Caractères alphabétiques et numériques")
print("3 - Caractères alphabétiques, numériques et caractères spéciaux")
choix = input("Votre choix (1, 2 ou 3) : ")

size = 15

groupes = []

if choix == "1":
    groupes = [uppercase_letters, lowercase_letters]
elif choix == "2":
    groupes = [uppercase_letters, lowercase_letters, digits]
elif choix == "3":
    groupes = [uppercase_letters, lowercase_letters, digits, special_characters]
else:
    print("Choix invalide ! Veuillez réessayer.")
    exit()

password = [random.choice(groupe) for groupe in groupes]
used_chars = ''.join(groupes)
password += random.choices(used_chars, k=size - len(password))

random.shuffle(password)

print("Mot de passe généré :", ''.join(password))
